### DISCUSSION TITLE
Why do the coordinates swap when doing matrix multiplication while rotating a rectangle?

### DISCUSSION THREAD

**Discussion by brandxntu** [pymupdf/PyMuPDF#1145] [Looking for help] [answered]
I am applying the derotation matrix to a rectangle on a page that is rotated 270 degrees, however I am noticing some strange behavior in the output of that rectangle. 

In this case, the derotation matrix is Matrix(0, 1, -1, 0, h, 0) 
(h corresponding to the height of the page)
For example if the rectangle I am using is  [x1 ,y1 ,x2, y2], multiplying the rectangle by the derotation matrix will give me an output of [h-y2, x1, h-y1, x2]. 

However, when doing the matrix multiplication by hand for each point, the result is: 

[x1, y1, 1] * Matrix(0, 1, -1, 0, h, 0)  = **[h-y1, x1]**

[x2, y2, 1] * Matrix(0, 1, -1, 0, h, 0) = **[h-y2, x2]**

While [h-y2, x1, h-y1, x2] and **[h-y1, x1, h-y2, x2]** are functionally the same rectangle (albeit with different corners), why is this swap of the x values done? Furthermore, how is this done mathematically in the code? What rule is used to swap these coordinates?

**Comment by JorjMcKie** [comment, marked as answer]
Your manual multiplication - corner point by corner point - leads to an "infinite" rectangle, see documentation.
The logic takes care that the result rect is defined by the diagonal of the points top-left and bottom-right.
To see the algorithm (which ultimately is written in C), lookup `fz_transform_rect` in MuPDF source code.
